File: tetris/src/main/java/com/example/tetris/Tetris.java
package com.example.tetris;

import com.fasterxml.jackson.databind.ObjectMapper;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

@WebServlet("/tetris")
public class Tetris extends HttpServlet {
    private static final long serialVersionUID = 1L;
    private Board board;
    private ObjectMapper mapper;

    @Override
    public void init() throws ServletException {
        board = new Board(10, 20); // Standard Tetris board size
        mapper = new ObjectMapper();
        
        // Example of placing a piece
        Piece piece = new Piece(0); // I piece
        if (board.canMove(piece, 3, 0)) {
            board.placePiece(piece, 3, 0);
        }
    }

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        if (request.getPathInfo() != null && request.getPathInfo().equals("/board")) {
            response.setContentType("application/json");
            response.getWriter().write(mapper.writeValueAsString(board.getBoard()));
        } else {
            response.setContentType("text/html");
            response.getWriter().println("<html><body><h1>Tetris Game</h1><canvas id='tetrisCanvas' width='300' height='600'></canvas><script src='tetris.js'></script></body></html>");
        }
    }
}

File: tetris/src/main/java/com/example/tetris/tetris.js
document.addEventListener("DOMContentLoaded", () => {
    const canvas = document.getElementById('tetrisCanvas');
    const context = canvas.getContext('2d');
    const scale = 30;

    function drawBoard(board) {
        context.clearRect(0, 0, canvas.width, canvas.height);
        for (let y = 0; y < board.length; y++) {
            for (let x = 0; x < board[y].length; x++) {
                if (board[y][x] !== 0) {
                    context.fillStyle = 'blue';
                    context.fillRect(x * scale, y * scale, scale, scale);
                    context.strokeRect(x * scale, y * scale, scale, scale);
                }
            }
        }
    }

    function updateBoard() {
        fetch('/tetris/board')
            .then(response => response.json())
            .then(data => {
                drawBoard(data);
            });
    }

    setInterval(updateBoard, 1000);
});
